src/main/flight/altitude.c

```diff
@@ -288,14 +244,27 @@ void calculateEstimatedAltitude(timeUs_t currentTimeUs)
     }
 #endif // USE_BARO
 
-    // apply Complimentary Filter to keep the calculated velocity based on baro velocity (i.e. near real velocity).
-    // By using CF it's possible to correct the drift of integrated accZ (velocity) without loosing the phase, i.e without delay
-    vel = vel * CONVERT_PARAMETER_TO_FLOAT(barometerConfig()->baro_cf_vel) + baroVel * (1.0f - CONVERT_PARAMETER_TO_FLOAT(barometerConfig()->baro_cf_vel));
     int32_t vel_tmp = lrintf(vel);
 
+    if (!ARMING_FLAG(ARMED)) {
+        baroAltOffset = baroAlt;
+        gpsAltOffset = gpsAlt;
+    }
+    baroAlt -= baroAltOffset;
+    gpsAlt -= gpsAltOffset;
+    
+    if (haveGPSAlt && haveBaroAlt) {
+        estimatedAltitude = gpsAlt * gpsTrust + baroAlt * (1-gpsTrust);
+    } else if (haveGPSAlt && !haveBaroAlt) {
+        estimatedAltitude = gpsAlt;
+    }
```

Comment:
What happens when both `haveGPSAlt` and `haveBaroAlt` are false? How does the code that will use `estimatedAltitude` know that altitude is no longer valid?